sgx-tools/dcap_linux.go

```diff
@@ -0,0 +1,511 @@
+	return rdata, nil
+}
+
+func DumpTargetInfo(targetInfo []byte) error {
+	ti := &intelsgx.Targetinfo{}
+	if err := restruct.Unpack(targetInfo, binary.LittleEndian, &ti); err != nil {
+		return err
+	}
+
+	logrus.Debugf("Quoting Enclave's TARGETINFO:\n")
+	logrus.Debugf("  Enclave Hash:       0x%v\n",
+		hex.EncodeToString(ti.Measurement[:]))
+	logrus.Debugf("  Enclave Attributes: 0x%v\n",
+		hex.EncodeToString(ti.Attributes[:]))
+	logrus.Debugf("  CET Attributes:     %#02x\n",
+		ti.CetAttributes)
+	logrus.Debugf("  Config SVN:         %#04x\n",
+		ti.ConfigSvn)
+	logrus.Debugf("  Misc Select:        %#08x\n",
+		ti.MiscSelect)
+	logrus.Debugf("  Config ID:          0x%v\n",
+		hex.EncodeToString(ti.ConfigId[:]))
+
+	return nil
+}
+
+func DumpLocalReport(report []byte) error {
+	r := &intelsgx.Report{}
+	if err := restruct.Unpack(report, binary.LittleEndian, &r); err != nil {
+		return err
+	}
+
+	logrus.Debugf("REPORT:")
+	logrus.Debugf("  CPU SVN:                        0x%v\n",
+		hex.EncodeToString(r.CpuSvn[:]))
+	logrus.Debugf("  Misc Select:                    %#08x\n",
+		r.MiscSelect)
+	logrus.Debugf("  Product ID:                     0x%v\n",
+		hex.EncodeToString(r.IsvExtProdId[:]))
+	logrus.Debugf("  Attributes:                     0x%v\n",
+		hex.EncodeToString(r.Attributes[:]))
+	logrus.Debugf("  Enclave Hash:                   0x%v\n",
+		hex.EncodeToString(r.MrEnclave[:]))
+	logrus.Debugf("  Enclave Signer:                 0x%v\n",
+		hex.EncodeToString(r.MrSigner[:]))
+	logrus.Debugf("  Config ID:                      0x%v\n",
+		hex.EncodeToString(r.ConfigId[:]))
+	logrus.Debugf("  ISV assigned Produdct ID:       %#04x\n",
+		r.IsvProdId)
+	logrus.Debugf("  ISV assigned SVN:               %d\n",
+		r.IsvSvn)
+	logrus.Debugf("  Config SVN:                     %#04x\n",
+		r.ConfigSvn)
+	logrus.Debugf("  ISV assigned Product Family ID: 0x%v\n",
+		hex.EncodeToString(r.IsvFamilyId[:]))
+	logrus.Debugf("  Report Data:                    0x%v\n",
+		hex.EncodeToString(r.ReportData[:]))
+
+	return nil
+}
+
+func DumpQuote(quote []byte) error {
+	q := &intelsgx.Quote{}
+	if err := restruct.Unpack(quote, binary.LittleEndian, &q); err != nil {
+		return err
+	}
+
+	logrus.Debugf("QUOTE:")
+	logrus.Debugf("  Version:                              %d\n",
+		q.Version)
+	logrus.Debugf("  Signature Type:                       %d\n",
+		q.SignatureType)
+	logrus.Debugf("  Gid:                                  %#08x\n",
+		q.Gid)
+	logrus.Debugf("  ISV assigned SVN for Quoting Enclave: %d\n",
+		q.ISVSvnQe)
+	logrus.Debugf("  ISV assigned SVN for PCE:             %d\n",
+		q.ISVSvnPce)
+	logrus.Debugf("  Base name:                            0x%v\n",
+		hex.EncodeToString(q.Basename[:]))
+	logrus.Debugf("  Report:                               ...\n")
+	logrus.Debugf("  Signature Length:                     %d\n",
+		q.SigLen)
+
+	return nil
+}
+
+func InitQuoteExRequest(AttkeyId []byte, bPubKeyId bool, bufSize uint64) (uint64, []byte, error) {
```

Comment:
The function name should be not exported, e.g, initQuoteExRequest. This will enforce you considering to design a high level API to encapsulate this low level function.